### ISSUE TITLE
aws-auth cache file not found (upgrade blocker)

### ISSUE THREAD

**Issue by galund** [alphagov/aws-auth#1]
`aws-auth` is not compatible with the latest version of the AWS CLI (I was trying 1.14.20).

Looks like the naming convention for files stored in the cache has changed: aws-auth is building the file name as follows, which no longer works:
```
cache_file=${CACHE_DIR}/${AWS_PROFILE}--$(echo "$role_arn" | sed 's/:/_/g' | sed 's/\//-/g').json
```

STR:
1) install AWS CLI 1.14+
2) remove cache files i.e. `rm -rf ~/.aws/cli/cache`
3) run e.g. `AWS_PROFILE=sops aws-auth`

The error I got when trying to use sops to get at our credentials store was:
```
$ aws-auth 
cat: ~/.aws/cli/cache/sops--arn_aws_iam__398263320410_role-sops-credentials-access.json: No such file or directory

**Comment by bandesz** [comment]
@galund I updated my AWS cli to 1.14.20 and it still works for me. Was the cache file created at all? If not then it should be an issue with the authentication and the error message might be swallowed.

If the cache file was created I suspect this might be a lower/uppercase issue with the profile name or the role name.

**Comment by bandesz** [comment]
@galund I deleted my cache files and was able to reproduce the issue. The AWS cli generates now a hash for the cache file name. As I'm not using this tool anymore (and also PaaS uses its own script) I probably won't be able to fix this.

**Comment by galund** [comment]
That's fine: if long-term DM continues to use this script then we'll maintain it. It might be interesting for us to compare the PaaS equivalent if there is one - can you point me at it? thanks!

**Comment by galund** [comment]
Couple of alternative approaches for comparison:
https://github.com/alphagov/paas-cf/blob/master/scripts/create_sts_token.sh
https://github.com/opsidian/awsc

The Notify team are also using this script.

**Comment by nwalke** [comment]
Just to collect more thoughts in one place, here's the botocore code that generates the name of the file: https://github.com/boto/botocore/blob/79589c66ae79572da2b33aeda58e3ac9e543ef61/botocore/credentials.py#L661

**Comment by risicle** [comment]
Yes it's looking like it would be quite painful to reliably generate that hash (although in a pinch it's possible to create a symlink from the "missing" filename to the hash-named file so long as it's easy to identify *which* is the right file). And even if that were solved, there's no guarantee the format of the file itself will remain stable. Probably best to look at other solutions.

**Comment by allait** [comment]
I've made a  small patch (#2) for this that seems to work. I didn't bother to add backwards compatibility, but if anyone has a use case where they need the script to work with both old and new awscli versions simultaneously then I could make the script fall back to the old file name if the new one doesn't exist.
